server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Plan.hs

```diff
@@ -211,15 +227,61 @@ validateVariables pgExecCtx variableValues = do
       res <- liftIO $ runExceptT (runLazyTx' pgExecCtx tx)
       liftEither res
 
-    -- Explicitly look for the class of errors raised when the format of a value provided
-    -- for a type is incorrect.
+    -- Explicitly look for the class of errors raised when the format of a value
+    -- provided for a type is incorrect.
     dataExnErrHandler = mkTxErrorHandler (has _PGDataException)
 
--- -------------------------------------------------------------------------------------------------
+-- | Internal: Used to collect information about various parameters
+-- of a subscription field's AST as we resolve them to SQL expressions.
+data QueryParametersInfo
+  = QueryParametersInfo
+  { _qpiReusableVariableValues     :: !GV.ReusableVariableValues
+  , _qpiSyntheticVariableValues    :: !(Seq (WithScalarType PGScalarValue))
+  , _qpiReferencedSessionVariables :: !(Set.HashSet SessVar)
+  -- ^ The session variables that are referenced in the query root fld's AST.
+  -- This information is used to determine a cohort's required session
+  -- variables
```

Comment:
I think it would be nice to pull the Haddock comment on `_cvSessionVariables` out into a `Note` and reference it in three places: the definition of `_cvSessionVariables`, the definition of `mkCohortVariables`, and the definition of `_qpiReferencedSessionVariables` (here). That way it’s much clearer what this is all about, but you don’t have to repeat yourself.